src/fields/field.impl.hh

```diff
@@ -436,8 +405,8 @@ void Field<spacedim,Value>::check_initialized_region_fields_() {
     	std::string region_list;
     	// has to deal with fact that reader can not deal with input consisting of simple values
     	string default_input=input_default();
-    	auto input_type = get_input_type();
-        Input::ReaderToStorage reader( default_input, input_type, Input::FileFormat::format_JSON );
+    	auto input_type = get_input_type().make_instance().first;
```

Comment:
Repeated call to make_instance() is not explicitly guaranteed to return exactly the same type. Remember types produced by instances and return them on second call.